crates/rust-analyzer/src/bin/logger.rs

```diff
@@ -57,22 +57,21 @@ impl Log for Logger {
                     record.module_path().unwrap_or_default(),
                     record.args(),
                 );
-                self.no_buffering
+
+                if self.no_buffering {
+                    self.flush();
```

Comment:
```suggestion
                    w.lock().flush();
```

(or hoist that to a local).

Then `flush` is dead code.

Reply:
Well we can't remove flush as it's part of Log trait.
Instead, I will put the actual flush inside of it.

Reply:
Ah, sorry, I missed that.

Reply:
Not to be too nit-picky, but this feels strange -- what if `log` calls `flush` internally?

I would have left `flush` unchanged, moved `w.lock()` to a variable, then called `flush` on that in this branch.

Reply:
I guess what you're suggesting makes sense, that way we had to acquire the lock twice.